index.js

```diff
@@ -78,7 +78,20 @@ function Hook (modules, options, onrequire) {
         }
       }
     } else {
-      filename = Module._resolveFilename(id, this)
+      try {
+        filename = Module._resolveFilename(id, this)
+      } catch (resolveErr) {
+        // If someone *else* monkey-patches before this monkey-patch, then that
+        // code might expect `require(someId)` to get through so it can be
+        // handled, even if `someId` cannot be resolved to a filename. In this
+        // case, instead of throwing we defer to the underlying `require`.
+        //
+        // For example the Azure Functions Node.js worker module does this,
+        // where `@azure/functions-core` resolves to an internal object.
+        // https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.5.2/src/setupCoreModule.ts#L46-L54
+        debug(`Module._resolveFilename('${id}') threw ${JSON.stringify(resolveErr.message)}, calling original Module.require`)
```

Comment:
From a performance point of view it's better to use the `%s` string replacement thingy instead of a template string as the former will only require processing if `debug` is enabled. Even better, we can use `%j` instead of `JSON.stringify`. That way the V8 engine doesn't have to spend a lot of time formatting the string unless it actually needs to be output to the console.

```suggestion
        debug('Module._resolveFilename('%s') threw %j, calling original Module.require', id, resolveErr.message)
```